astroid/nodes/node_classes.py

```diff
@@ -1524,58 +1524,34 @@ class Comprehension(NodeNG):
     'for x in some_values'
     """
 
-    _astroid_fields = ("target", "iter", "ifs")
-    _other_fields = ("is_async",)
-
-    optional_assign = True
-    """Whether this node optionally assigns a variable."""
-
-    lineno: None
-    col_offset: None
-    end_lineno: None
-    end_col_offset: None
+    target: NodeNG
+    """What is assigned to by the comprehension."""
 
-    def __init__(self, parent: NodeNG | None = None) -> None:
-        """
-        :param parent: The parent node in the syntax tree.
-        """
-        self.target: NodeNG | None = None
-        """What is assigned to by the comprehension."""
+    iter: NodeNG
+    """What is iterated over by the comprehension."""
 
-        self.iter: NodeNG | None = None
-        """What is iterated over by the comprehension."""
+    ifs: list[NodeNG]
+    """The contents of any if statements that filter the comprehension."""
 
-        self.ifs: list[NodeNG] = []
-        """The contents of any if statements that filter the comprehension."""
+    is_async: bool
+    """Whether this is an asynchronous comprehension or not."""
 
-        self.is_async: bool | None = None
-        """Whether this is an asynchronous comprehension or not."""
+    _astroid_fields = ("target", "iter", "ifs")
```

Comment:
Please keep these at the top. It makes it easier to `git blame` these lines to an actual change to them.